### PR TITLE
chore: remove unnecessary code

### PR DESCRIPTION
**Pull request by lvimuser**
# Description

Just a simple cleanup. The deepcopy can be removed, and the component logic was duplicated.

### CODE REVIEW

File: lua/lvim/lsp/config.lua [hunk 1]
```diff
@@ -64,12 +64,11 @@ return {
       header = "",
       prefix = "",
       format = function(d)
-        local t = vim.deepcopy(d)
```
Comment (kylo252) [inline lua/lvim/lsp/config.lua:67]: the `deepcopy()` was necessary to avoid messing up the original diagnostics, are you sure it's not needed anymore?

Comment (lvimuser) [reply lua/lvim/lsp/config.lua:67]: we're not mutating anymore though, instead of changing the field we just simply return.

Comment (kylo252) [reply lua/lvim/lsp/config.lua:67]: nice catch!

File: lua/lvim/core/lualine/components.lua [hunk 2]
```diff
@@ -47,15 +47,10 @@ return {
     function()
       local utils = require "lvim.core.lualine.utils"
       if vim.bo.filetype == "python" then
-        local venv = os.getenv "CONDA_DEFAULT_ENV"
+        local venv = os.getenv "CONDA_DEFAULT_ENV" or os.getenv "VIRTUAL_ENV"
```
[the remark this answers is not in the document]
Comment (lvimuser) [reply lua/lvim/core/lualine/components.lua:50]: Does `vim.env` keeps track of changes? (e.g. you may change venv without restarting the editor...). I'd rather not change it (or leave it for another PR).

[the remark this answers is not in the document]
Comment (kylo252) [reply lua/lvim/core/lualine/components.lua:50]: nvm, `vim.env` is implemented with `vim.fn.getenv` so it's probably slower

**Review by kylo252 — APPROVED** [review]
LGTM! 🚀

**Pull request merged**